content/en/docs/collector/internal-telemetry.md

```diff
@@ -25,7 +27,7 @@ endpoint to one specific or all network interfaces when needed. For
 containerized environments, you might want to expose this port on a public
 interface.
 
-Set the address in the config `service::telemetry::metrics`:
+Set the address in the config `service.telemetry.metrics`:
```

Comment:
Why this change? `service::telemetry::metrics` actually works, while `service.telemetry.metrics` does not, that is to say `key::subkey: value` is equivalent to

```yaml
key:
  subkey: value
```

I think we should prefer the `::` variant because of that

```suggestion
Set the address in the config `service::telemetry::metrics`:
```

Reply:
> Why this change?

Because `service::telemetry::metrics` isn't valid formal YAML syntax or even a recognized informal YAML syntax.

> `service::telemetry::metrics` actually works, 

What do you mean "it works"? Where, in which context or language?

> ... that is to say `key::subkey: value` is equivalent to

Are you claiming that that is official YAML syntax? If so, I'm not seeing it in the [YAML spec](https://yaml.org/spec/1.2.2/).

Reply:
Ooooh, I see that the double-colon notation is accepted by `otelcol --set`. Hmm, given that the prose is referring to YAML config, I'd suggest using the dot notation in the prose. WDYT?

Reply:
If you feel strongly about showing the double-colon notation, we should make it clear that it's the `otelcol --set` syntax.

But given that the set command has limitations (from the config page "Note tha the `--set` option doesn't support setting a key that contains a dot or an equal sign"), it might be better to stick to illustrating the YAML config syntax only?

Reply:
> Ooooh, I see that the double-colon notation is accepted by otelcol --set. Hmm, given that the prose is referring to YAML config, I'd suggest using the dot notation in the prose. WDYT?

@chalin Not only `--set`, this also works in the Collector configuration files. The following is a valid snippet of Collector configuration:

```yaml
service:
   telemetry::metrics::level: detailed
```

which is equivalent to:

```yaml
service:
  telemetry:
    metrics:
      level: detailed
```

Reply:
@mx-psi - ok, from [Collector Configuration], I see that `--set` and `--config="yaml:..."` accept the double-colon syntax.

As for the use of keys containing double colons (like the example you gave above) that are reinterpreted as nested structures, that is news to me and it isn't documented in [Collector Configuration] page, nor have I seen any example in the docs or elsewhere. But I'm willing to believe you.

Can you show me config files that use this syntax? 

The [Collector Configuration] page needs to be updated to reflect this non-standard interpretation of YAML keys then.

[Collector Configuration]: https://opentelemetry.io/docs/collector/configuration/

Reply:
I read through the [Collector Configuration](https://opentelemetry.io/docs/collector/configuration/) page, and it does demonstrate the use of double colons when passing config settings at the command line:

> You can also provide configurations using environment variables, YAML paths, or HTTP URIs. For example:
> 
> `otelcol --config=env:MY_CONFIG_IN_AN_ENVVAR --config=https://server/config.yaml`
> `otelcol --config="yaml:exporters::debug::verbosity: normal"`

I propose to revert the paths in this PR to the double colon so we can move the PR forward. I will also create a new issue for adding more details about this syntax to the Configuration page.

WDYT?

Reply:
Sounds good to me. Pls cross-link the new issue here. Thanks!

Reply:
See issue #4484.